.github/actions/repo-gardening/src/tasks/add-labels/index.js

```diff
@@ -0,0 +1,166 @@
+ */
+async function getFiles( octokit, owner, repo, number ) {
+	const fileList = [];
+
+	debug( 'add-labels: Get list of files modified in this PR.' );
+
+	for await ( const response of octokit.paginate.iterator( octokit.pulls.listFiles, {
+		owner,
+		repo,
+		pull_number: +number,
+	} ) ) {
+		response.data.map( file => {
+			fileList.push( file.filename );
+		} );
+	}
+
+	return fileList;
+}
+
+/**
+ * Clean up a feature name:
+ * - Replace dashes by spaces.
+ * - Capitalize.
+ *
+ * @param {string} name - Feature name.
+ *
+ * @returns {string} Cleaned up feature name.
+ */
+function cleanName( name ) {
+	return (
+		name
+			// Break up words
+			.split( '-' )
+			// Capitalize first letter of each word.
+			.map( word => `${ word[ 0 ].toUpperCase() }${ word.slice( 1 ) }` )
+			// Spaces between words.
+			.join( ' ' )
+	);
+}
+
+/**
+ * Build a list of labels to add to the issue, based off our file list.
+ *
+ * @param {GitHub} octokit - Initialized Octokit REST client.
+ * @param {string} owner   - Repository owner.
+ * @param {string} repo    - Repository name.
+ * @param {string} number  - PR number.
+ *
+ * @returns {Promise<Array>} Promise resolving to an array of keywords we'll search for.
+ */
+async function getLabelsToAdd( octokit, owner, repo, number ) {
+	const keywords = new Set();
+
+	// Get next valid milestone.
+	const files = await getFiles( octokit, owner, repo, number );
+
+	if ( ! files ) {
+		throw new Error( 'No files were modified in this PR' );
+	}
+
+	debug( 'add-labels: Loop through all files modified in this PR and add matching labels.' );
+
+	files.map( file => {
+		// Plugins.
+		const plugin = file.match( /^projects\/plugins\/(?<plugin>\w*)\// );
+		const pluginName = plugin && plugin.groups.plugin;
+		if ( pluginName ) {
+			keywords.add( `[Plugin] ${ cleanName( pluginName ) }` );
+		}
+
+		// Packages.
+		const packages = file.match( /^projects\/packages\/(?<package>\w*)\// );
+		const packageName = packages && packages.groups.package;
+		if ( packageName ) {
+			keywords.add( `[Package] ${ cleanName( packageName ) }` );
+			keywords.add( `[Status] Needs Package Release` );
+		}
+
+		// Modules.
+		const module = file.match(
+			/^projects\/plugins\/jetpack\/?(?<test>tests\/php\/)?modules\/(?<module>[a-zA-Z-]*)\//
+		);
+		const moduleName = module && module.groups.module;
+		if ( moduleName ) {
+			keywords.add( `${ cleanName( moduleName ) }` );
+		}
+		if ( module && module.groups.test ) {
+			keywords.add( 'Unit Tests' );
+		}
+
+		// Actions.
+		const actions = file.match(
+			/^\.github\/actions\/|projects\/github-actions\/(?<action>[a-zA-Z-]*)\//
+		);
+		if ( actions !== null ) {
+			keywords.add( 'Actions' );
+			const actionName = actions && actions.groups.action;
+			if ( actionName ) {
+				keywords.add( `[Actions] ${ cleanName( actionName ) }` );
+			}
+		}
+
+		// Docker.
+		const docker = file.match( /^tools\/docker\// );
+		if ( docker !== null ) {
+			keywords.add( 'Docker' );
+		}
+
```

Comment:
Could I be selfish and get `/^tools\/cli\//` added for `[Tools] Development CLI`?

Reply:
Done in 7e1a88cd0e66f4c3ae84598ae291b9b242688f8e